CHANGELOG.md

```diff
@@ -7,6 +7,76 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [0.9.0] - TBD
+
+### Known Issues
+
+- `2D` and `3D` in generated `snake_case` names are delimited incorrectly. e.g. `position2_d`.
+
+- There are no ways to perform certain operations on `Dictionary` and `VariantArray` with the `Shared` typestate, even in `unsafe` contexts.
+
+- The bindings does not support Godot 4.0 (`master` branch) yet.
+
+### Added
+
+- All public functions now have the `#[inline]` attribute, improving cross-crate inlining in builds without LTO.
+
+- A curated `prelude` module is added in `gdnative` crate, containing common types and traits.
+
+- Added the `SubClass` trait, which allows for static up-casts and static validation of downcasts.
+
+- Added the `OwnedToVariant` trait and derive macro which enabled `Variant` conversion for more types.
+
+- The `NativeScript` and `#[methods]` proc-macros now report errors more accurately.
+
+### Changed
+
+- **The default API version is now Godot 3.2.2-stable.**
+
+- The object reference system is revamped using the typestate pattern, with semantics that model Godot behavior more accurately, allowing for clearer boundaries between safe and unsafe code.
+
+- API methods are now generic over types that can be converted to `Variant`, `GodotString`, or generated API types, removing the need for boilerplate code like `&thing.into()`.
+
+- Enums in the API are now represented more accurately as newtypes around `i64`. They are available in the same modules where their associated objects are defined.
+
+- `Dictionary` and `VariantArray` now use the typestate pattern as well.
+
+- The typed arrays are unified into a generic `TypedArray` type. The old names remain as type aliases.
+
+- Moved generated bindings into the `gdnative::api` module, making the top-level namespace easier to navigate.
+
```

Comment:
I'd add that it's now possible to create custom bindings crates without having to fork the repository since `Api::new()` takes all the data as an argument now? #480

Reply:
Added!